msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -345,13 +365,25 @@ private MsalAccessTokenCacheItem FindAccessTokenCommon
                 }
 
                 requestParams.RequestContext.Logger.Info("Matching entry count -" + tokenCacheItems.Count);
-
-                IEnumerable<MsalAccessTokenCacheItem> filteredItems =
-                    tokenCacheItems.Where(
-                            item =>
-                                item.ScopeSet.ScopeContains(requestParams.Scope));
-
-                requestParams.RequestContext.Logger.Info("Matching entry count after filtering by scopes - " + filteredItems.Count());
+    
+
+                IEnumerable<MsalAccessTokenCacheItem> filteredItems =
+                   tokenCacheItems.Where(
+                           item => item.ScopeSet.ScopeContains(requestParams.Scope));
```

Comment:
Should then only do this in case it's not ADFS.... seems like waste of CPU cycles.

Reply:
The MSAL library adds the scopes in this format if none are present in the response; shouldn't happen, but technically is a possibility